cscs-checks/apps/amber/amber_check.py

```diff
@@ -58,102 +59,55 @@
 }
 
 
-class AmberBaseCheck(rfm.RunOnlyRegressionTest):
+def my_param_filter(inherited_params, match):
+    '''Filter out variants by inspecting the parameter's first element.'''
+    filtered_params = []
+    for p in inherited_params:
+        if match in p[0]:
+            filtered_params.append(p)
+    return tuple(filtered_params)
+
+
+def inherit_cpu_only(p):
+    return my_param_filter(p, 'cpu')
+
+
+def inherit_gpu_only(p):
+    return my_param_filter(p, 'gpu')
```

Comment:
I think that you can easily replace these with a single line lambda in the parameter definitions:

```python
    platform = parameter(
        inherit_params=True,
        filter_params=lambda params: p for p in params if p == 'gpu'
    )

Reply:
As far as I know, `for` and `if` statements cannot be part of `lambda` functions. I found a way to shorten all those awful functions a bit with a `filter`, but 2 still had to be kept. Maybe you know how else this can be improved? I tried using the `filter` directly in the `parameter`, but its syntax made it difficult.